miner/block_builder.go

```diff
@@ -177,30 +178,60 @@ func calcHdistRange(id types.LayerID, hdist types.LayerID) (bottom types.LayerID
 	return bottom, top
 }
 
-func (t *BlockBuilder) createBlock(id types.LayerID, atxID types.AtxId, eligibilityProof types.BlockEligibilityProof,
-	txids []types.TransactionId, atxids []types.AtxId) (*types.Block, error) {
-
+func (t *BlockBuilder) getVotes(id types.LayerID) ([]types.BlockID, error) {
 	var votes []types.BlockID = nil
-	var err error
+
+	// if genesis
 	if id == config.Genesis {
 		return nil, errors.New("cannot create blockBytes in genesis layer")
-	} else if id == config.Genesis+1 {
-		votes = append(votes, config.GenesisId)
-	} else { // get from hare
-		bottom, top := calcHdistRange(id, t.hdist)
-		votes, err = t.hareResult.GetResult(bottom, top)
-		if err != nil {
-			t.With().Warning("Could not get hare result during block creation",
-				log.Uint64("bottom", uint64(bottom)), log.Uint64("top", uint64(top)),
-				log.Uint64("hdist", uint64(t.hdist)), log.Err(err))
+	}
+
+	// if genesis+1
+	if id == config.Genesis+1 {
+		return append(votes, config.GenesisId), nil
+	}
+
+	// not genesis, get from hare
+	bottom, top := calcHdistRange(id, t.hdist)
+
+	votesBottom, err := t.hareResult.GetResult(bottom, bottom)
+	if votesBottom == nil || len(votesBottom) == 0 {
```

Comment:
Are the 2 options possible?

Reply:
It doesn't matter. I don't count on how GetResult is implemented. I play it defensive.